File: PLP_python/Aula03_ListaDeExercicios(15)/Q13_Matematica.py
class Q013_Matematica:

    @staticmethod
    def fatorial(n):
        if n == 0 or n == 1:
            return 1
        else:
            return n * Q013_Matematica.fatorial(n - 1)


if __name__ == "__main__":
    print("O fatorial de 5 é: {}".format(Q013_Matematica.fatorial(5)))
    print("O fatorial de 0 é: {}".format(Q013_Matematica.fatorial(0)))

File: PLP_Java/src/Aula03_ListaDeExercicio/Q13_Matematica.java
package Aula03_ListaDeExercicio;

class Q013_Matematica {

    public static int fatorial(int n) {
        if (n == 0 || n == 1) {
            return 1;
        } else {
            return n * fatorial(n - 1);
        }
    }

    public static void main(String[] args) {
        System.out.println("O fatorial de 5 é: " + Q013_Matematica.fatorial(5));
        System.out.println("O fatorial de 0 é: " + Q013_Matematica.fatorial(0));
    }
}



